app/parse.py

```
@@ -15,8 +29,106 @@ class Product:
     num_of_reviews: int
 
 
+PRODUCT_NAME = [field.name for field in fields(Product)]
+
+
+def get_driver() -> webdriver:
+    options = webdriver.ChromeOptions()
+    options.add_argument("--headless")
+
+    driver = webdriver.Chrome(options=options)
+    return driver
+
+
+def cookies(driver: webdriver) -> None:
+    try:
+        wait = WebDriverWait(driver, 15)
+        accept_cookie = wait.until(
+            EC.element_to_be_clickable((
+                By.CLASS_NAME, "acceptCookies"
+            ))
+        )
+
+        accept_cookie.click()
+        time.sleep(5)
+    except TimeoutException:
+        print("cookie doesn't found")
+
+
+def scroll_more(driver: webdriver) -> None:
+    while True:
+        try:
+            wait = WebDriverWait(driver, timeout=15)
+            more = wait.until(
+                EC.element_to_be_clickable(
+                    (
+                        By.CLASS_NAME, "ecomerce-items-scroll-more")
+                )
+            )
+            more.click()
+
+        except Exception:
+            break
```

Pagination (`scroll_more`) is brittle and hides errors:
- It clicks the “more” button repeatedly but doesn't verify that new items are loaded.
- It catches any `Exception`, which can swallow programming errors.
Improve by tracking the number of `.thumbnail` elements before clicking, clicking, then waiting (with WebDriverWait) until the count increases. Stop when the button is no longer present/clickable or the count does not increase. Catch specific exceptions only (e.g., `TimeoutException`, `NoSuchElementException`) instead of a broad `Exception`.